climada/util/coordinates.py

```diff
@@ -1591,20 +1604,16 @@ def read_raster_bounds(path, bounds, res=None, bands=None):
             res = (src.transform[0], src.transform[4])
         res = (np.abs(res[0]), np.abs(res[1]))
 
-        width, height = bounds[2] - bounds[0], bounds[3] - bounds[1]
-        shape = (int(np.ceil(height / res[1]) + 1),
-                 int(np.ceil(width / res[0]) + 1))
+        # make sure that the extent of pixel centers covers the specified region
+        bounds = (bounds[0] - res[0], bounds[1] - res[1], bounds[2] + res[0], bounds[3] + res[1])
```

Comment:
Could this not interfere with the sign of the res?

Reply:
No, the sign of `res` is ignored (`np.abs` is applied a few lines above this line). As explained in the docstring, the orientation is always inferred from the data set's axis orientations.

Reply:
Good point, I just added unit tests to make sure that this is working as expected.